### PR TITLE
Added date validation handling to data management UI. (Pt-4)

### PR DESCRIPTION
**Pull request by 6s-1**
## 📝 Description

Adds date validation to the Data Management page so users can only enter valid dates and relationships.

## Changes
- **Date format:** Only `dd-mm-yyyy` is accepted; invalid input shows an error message.
- **Project dates:** Last modified must be after date created; otherwise an error is shown.
- **Skill dates:** Skill date must be within the project’s created–last modified range; otherwise an error is shown.
- **UI:** Error messages are shown in a styled alert above the projects table.
- **Tests:** New test for date format validation error.

**Closes:** #823 #824 #825 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [x] ♻️ Refactoring
- [x] ⚡ Performance improvement

---

## 🧪 Testing

-  **Date format validation:** Edit a project or skill date, enter invalid formats (e.g. `invalid`, `12/25/2024`, `25-13-2024`), blur or press Enter → error message appears.
-  **Valid format:** Enter valid `dd-mm-yyyy` (e.g. `25-12-2024`) → saves without error.
-  **Modified > created:** Set last modified before or equal to created → error message appears.
-  **Skill in range:** Set skill date outside project range → error message appears.
-  **Hub → Data Management:** From Hub, click Data Management card → navigates to Data Management page.
-  **Unit tests:** `cd desktop && npm test -- DataManagementPage.test` passes.

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [x] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules
- [x] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

All tests pass:

Screenshot 2026-03-14 at 6 41 55 PM

UI error alerts and conditions:

Screenshot 2026-03-14 at 6 36 02 PM
Screenshot 2026-03-14 at 6 34 27 PM
Screenshot 2026-03-14 at 6 34 11 PM

### CODE REVIEW

File: desktop/src/pages/DataManagementPage.tsx [hunk 1]
```diff
@@ -140,7 +150,17 @@ export function DataManagementPage() {
...
 
-  const handleUpdateSkillDate = async (skillId: number, date: string) => {
+  const handleUpdateSkillDate = async (
+    skillId: number,
+    date: string,
+    projectSig: string
+  ) => {
+    setDateError(null);
+    const proj = projects.find((p) => p.project_signature === projectSig);
+    if (proj && (date < proj.created_at || date > proj.last_modified)) {
```
Comment (kjassani) [inline desktop/src/pages/DataManagementPage.tsx]: This range check is doing a raw string comparison, but project dates can include timestamps while the edited skill date is date-only. For example, \2026-01-15 compares as earlier than 2026-01-15T10:30:00Z, so a skill on the project’s created day gets rejected even though it should be valid. Could we consider normalizing both sides to the same date format before comparing, and add a test for the same-day boundary case.

Comment (6s-1) [reply desktop/src/pages/DataManagementPage.tsx]: Thanks for catching that, I have updated the same. All tests pass.

**Review by KarimKhalil33 — APPROVED** [review]
Great work shreya! code lgtm
